### ISSUE TITLE
Assert opening a .NET Core project under CPS DBG bits

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn-project-system#1296]
Get the following assert opening a .NET Core project:

```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
Passing the project full path during rule construction!  Instead, null should be passed so that the project file name will be automatically picked up.  Otherwise if the project is renamed, the file name passed can become stale.



   at Microsoft.Report.Fail(String message)
   at Microsoft.Report.If(Boolean condition, String message)
   at Microsoft.VisualStudio.ProjectSystem.PropertyPages.PageRuleBase..ctor(Rule rule, String file, String itemType, String itemName, PropertyPagesDataModelProvider provider) in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\PropertyPages\PageRuleBase.cs:line 111
   at Microsoft.VisualStudio.ProjectSystem.PropertyPages.PageRule..ctor(Rule rule, String file, String itemType, String itemName, PropertyPagesDataModelProvider provider) in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\PropertyPages\PageRule.cs:line 34
   at Microsoft.VisualStudio.ProjectSystem.PropertyPages.PropertyPagesDataModelProvider.GetRule(Rule schema, String file, String itemType, String itemName) in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\PropertyPages\PropertyPagesDataModelProvider.cs:line 139
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesProjectTreeProvider.GetRuleForUnresolvableReference(IProjectPropertiesContext unresolvedContext, IProjectCatalogSnapshot catalogs, ConfiguredProjectExports configuredProjectExports)
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesProjectTreeProvider.CreateProjectItemTreeNode(IProjectTree providerRootTreeNode, IDependencyNode nodeInfo, IProjectCatalogSnapshot catalogs)
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesProjectTreeProvider.CreateOrUpdateSubTreeProviderNode(IProjectTree dependenciesNode, IProjectDependenciesSubTreeProvider subTreeProvider, IDependenciesChangeDiff changes, IProjectCatalogSnapshot catalogs, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesProjectTreeProvider.<>c__DisplayClass22_0.<OnDependenciesChanged>b__0(IProjectVersionedValue`1 treeSnapshot, ConfiguredProjectExports configuredProjectExports, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.TreeUpdateSubmission.<UpdateTree>d__25.MoveNext() in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem\ProjectTree\Providers\ProjectTreeProviderBase.cs:line 1025
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.TreeUpdateSubmission.UpdateTree(IProjectVersionedValue`1 currentTree, ConfiguredProjectExports configuredProjectExports, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.<>c__DisplayClass95_1.<<Initialize>b__1>d.MoveNext() in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem\ProjectTree\Providers\ProjectTreeProviderBase.cs:line 464
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.<>c__DisplayClass95_1.<Initialize>b__1()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.RunAsync[T](Func`1 asyncMethod, Boolean synchronouslyBlocking, JoinableTaskCreationOptions creationOptions)
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.RunAsync[T](Func`1 asyncMethod)
   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.<>c__DisplayClass95_0.<<Initialize>b__0>d.MoveNext() in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem\ProjectTree\Providers\ProjectTreeProviderBase.cs:line 444
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.<>c__DisplayClass95_0.<Initialize>b__0(TreeUpdateSubmission submission)
   at System.Threading.Tasks.Dataflow.TransformBlock`2.ProcessMessageWithTask(Func`2 transform, KeyValuePair`2 messageWithId)
   at System.Threading.Tasks.Dataflow.TransformBlock`2.<>c__DisplayClass10.<.ctor>b__4(KeyValuePair`2 messageWithId)
   at System.Threading.Tasks.Dataflow.Internal.TargetCore`1.ProcessMessagesLoopCore()
   at System.Threading.Tasks.Dataflow.Internal.TargetCore`1.<ProcessAsyncIfNecessary_Slow>b__3(Object thisTargetCore)
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.Tasks.Task.ExecutionContextCallback(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)
   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()

---------------------------
Abort   Retry   Ignore   
---------------------------

```

**Comment by davkean** [comment]
Another one:
```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
Passing the project full path during rule construction!  Instead, null should be passed so that the project file name will be automatically picked up.  Otherwise if the project is renamed, the file name passed can become stale.



   at Microsoft.Report.Fail(String message)

   at Microsoft.Report.If(Boolean condition, String message)

   at Microsoft.VisualStudio.ProjectSystem.PropertyPages.PageRuleBase..ctor(Rule rule, String file, String itemType, String itemName, PropertyPagesDataModelProvider provider) in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\PropertyPages\PageRuleBase.cs:line 111

   at Microsoft.VisualStudio.ProjectSystem.PropertyPages.PageRule..ctor(Rule rule, String file, String itemType, String itemName, PropertyPagesDataModelProvider provider) in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\PropertyPages\PageRule.cs:line 34

   at Microsoft.VisualStudio.ProjectSystem.PropertyPages.PropertyPagesDataModelProvider.GetRule(Rule schema, String file, String itemType, String itemName) in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\PropertyPages\PropertyPagesDataModelProvider.cs:line 139

   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesProjectTreeProvider.GetRuleForResolvableReference(IProjectPropertiesContext unresolvedContext, KeyValuePair`2 resolvedReference, IProjectCatalogSnapshot catalogs, ConfiguredProjectExports configuredProjectExports, Boolean isGenericDependency)

   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesProjectTreeProvider.CreateProjectItemTreeNode(IProjectTree providerRootTreeNode, IDependencyNode nodeInfo, IProjectCatalogSnapshot catalogs)

   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesProjectTreeProvider.CreateOrUpdateSubTreeProviderNode(IProjectTree dependenciesNode, IProjectDependenciesSubTreeProvider subTreeProvider, IDependenciesChangeDiff changes, IProjectCatalogSnapshot catalogs, CancellationToken cancellationToken)

   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesProjectTreeProvider.<>c__DisplayClass22_0.<OnDependenciesChanged>b__0(IProjectVersionedValue`1 treeSnapshot, ConfiguredProjectExports configuredProjectExports, CancellationToken cancellationToken)

   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.TreeUpdateSubmission.<UpdateTree>d__25.MoveNext() in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem\ProjectTree\Providers\ProjectTreeProviderBase.cs:line 1025

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.TreeUpdateSubmission.UpdateTree(IProjectVersionedValue`1 currentTree, ConfiguredProjectExports configuredProjectExports, CancellationToken cancellationToken)

   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.<>c__DisplayClass95_1.<<Initialize>b__1>d.MoveNext() in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem\ProjectTree\Providers\ProjectTreeProviderBase.cs:line 464

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.<>c__DisplayClass95_1.<Initialize>b__1()

   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.RunAsync[T](Func`1 asyncMethod, Boolean synchronouslyBlocking, JoinableTaskCreationOptions creationOptions)

   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.RunAsync[T](Func`1 asyncMethod)

   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.<>c__DisplayClass95_0.<<Initialize>b__0>d.MoveNext() in E:\CPS\src\Microsoft.VisualStudio.ProjectSystem\ProjectTree\Providers\ProjectTreeProviderBase.cs:line 444

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.VisualStudio.ProjectSystem.ProjectTreeProviderBase.<>c__DisplayClass95_0.<Initialize>b__0(TreeUpdateSubmission submission)

   at System.Threading.Tasks.Dataflow.TransformBlock`2.ProcessMessageWithTask(Func`2 transform, KeyValuePair`2 messageWithId)

   at System.Threading.Tasks.Dataflow.TransformBlock`2.<>c__DisplayClass10.<.ctor>b__4(KeyValuePair`2 messageWithId)

   at System.Threading.Tasks.Dataflow.Internal.TargetCore`1.ProcessMessagesLoopCore()

   at System.Threading.Tasks.Dataflow.Internal.TargetCore`1.<ProcessAsyncIfNecessary_Slow>b__3(Object thisTargetCore)

   at System.Threading.Tasks.Task.InnerInvoke()

   at System.Threading.Tasks.Task.Execute()

   at System.Threading.Tasks.Task.ExecutionContextCallback(Object obj)

   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)

   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)

   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()

   at System.Threading.ThreadPoolWorkQueue.Dispatch()

   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()


---------------------------
Abort   Retry   Ignore   
---------------------------

```

**Comment by davkean** [comment]
It looks like this is going down this path in GetRuleForUnresolvableReference:
```
            var emptyRule = RuleExtensions.SynthesizeEmptyRule(unresolvedContext.ItemType);
            return configuredProjectExports.PropertyPagesDataModelProvider.GetRule(
                        emptyRule, 
                        unresolvedContext.File, 
                        unresolvedContext.ItemType, 
                        unresolvedContext.ItemName);
```

Looks like we don't have a rule for PackageReference? Does that look right @abpiskunov?

**Comment by abpiskunov** [comment]
I just debugged through it and those code paths not even hit in the debugger. It could be fixed by my latest change for returning properties for top level dependencies. I did change a little logic hwo we find xaml rules for dependency nodes. So now for our standard nodes we should not hit that path, it is there just in case there are some custom nodes that don't have any known xaml rules, or have too many (> than 1) BrowseObject rules.
